code/modules/games/cards.dm

```diff
@@ -214,7 +224,7 @@
 	if(!discarding || !to_discard[discarding] || !usr || !src) return
 
 	var/datum/playingcard/card = to_discard[discarding]
-	qdel(to_discard)
+	to_discard.Cut()
```

Comment:
I am assuming that this list was nulled for a reason previously (the qdel) so I've done the Cut() here. No idea why it needs to be, though, considering this proc doesn't involve spawn or waitfor=0.